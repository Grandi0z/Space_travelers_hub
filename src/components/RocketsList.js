import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import fetchRockets from './actions';

const RocketsList = () => {
  const [reserved, setReserved] = useState(false);
  const handleReserveClick = () => {
    setReserved(true);
  };
  const handleCancelReserveClick = () => {
    setReserved(false);
  };

  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rockets.rockets);

  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);

  return (
    <div>
      {rockets.map((rocket) => (
        <div className="rocket-container" key={rocket.id}>
          <div className="rocket-item-image">
            <img className="rocket-image" src={rocket.flickr_images[0]} alt={rocket.rocket_name} />
          </div>
          <div className="rocket-item">
            {reserved ? (<span className="reserved" id="reserved"> Reserved </span>) : ''}
            <span className="rocket-name">{rocket.name}</span>
            <p className="rocket-description">{rocket.description}</p>
            {reserved ? (<button type="button" className="cancel-button" id="cancelReserveBtn" onClick={handleCancelReserveClick}> Cancel Reserve </button>) : (
              <button type="button" className="reserve-button" id="reserveBtn" onClick={handleReserveClick}> Reserve Rocket </button>)}
          </div>
        </div>
      ))}
    </div>
  );
};

export default RocketsList;
