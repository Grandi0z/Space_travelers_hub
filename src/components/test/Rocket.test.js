import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import RocketsList from '../RocketsList';
import rocketsReducer from '../../redux/features/rockets/reducers';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
  },
});

describe('Rockets', () => {
  it('Should match the rocket snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}><RocketsList /></Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
