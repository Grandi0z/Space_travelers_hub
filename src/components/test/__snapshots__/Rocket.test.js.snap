// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Should match the snapshot 1`] = `
<div>
  No mission yet
</div>
`;

exports[`Rockets Should match the ro snapshot 1`] = `<div />`;

exports[`Rockets Should match the rocket snapshot 1`] = `<div />`;
